src/leap/mail/adaptors/soledad.py

```diff
@@ -483,17 +529,24 @@ def create(self, store):
 
         # TODO check that the doc_ids in the mdoc are coherent
         d = []
-        d.append(self.mdoc.create(store))
+        mdoc_created = self.mdoc.create(store)
+        d.append(mdoc_created)
         d.append(self.fdoc.create(store))
-        if self.hdoc.doc_id is None:
-            d.append(self.hdoc.create(store))
-        for cdoc in self.cdocs.values():
-            if cdoc.doc_id is not None:
-                # we could be just linking to an existing
-                # content-doc.
-                continue
-            d.append(cdoc.create(store))
-        return defer.gatherResults(d)
+
+        if not self._is_copy:
+            if self.hdoc.doc_id is None:
+                d.append(self.hdoc.create(store))
+            for cdoc in self.cdocs.values():
+                if cdoc.doc_id is not None:
+                    # we could be just linking to an existing
+                    # content-doc.
+                    continue
+                d.append(cdoc.create(store))
+
+        if notify_just_mdoc:
+            return mdoc_created
+        else:
+            return defer.gatherResults(d)
```

Comment:
Hmm. I see your point, but I'm afraid that if I catch the exceptions in here I'll be avoiding the failure to propagate to upper layers, and thus invalidating more meaningfull errBacks (at the level of IMAP server methods, for instance: there are always methods adding callbacks and errbacks to these soledad deferreds, and it makes more sense to catch errors at that level - notifying to the MUA that an append has failed, for instance).

I think I want to make some real tests to see how this would behave before taking action.